src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,10 @@
 
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
```

Comment:
Now we can use any type for T in our impls. But it does not make sense to have String or Object instead of T. As a solution we can restrict types using T extends .... in here